src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2813,8 +2813,21 @@ private void ReinferMethodAndVisitArguments(BoundCall node, TypeWithState receiv
                 method = (MethodSymbol)AsMemberOfType(receiverType.Type, method);
             }
 
-            method = VisitArguments(node, node.Arguments, refKindsOpt, method.Parameters, node.ArgsToParamsOpt,
-                node.Expanded, node.InvokedAsExtensionMethod, method).method;
+            ImmutableArray<VisitArgumentResult> results;
+            (method, results) = VisitArguments(node, node.Arguments, refKindsOpt, method.Parameters, node.ArgsToParamsOpt,
+                node.Expanded, node.InvokedAsExtensionMethod, method);
+
+            var arguments = node.Arguments;
```

Comment:
>var [](start = 12, length = 3)

Consider moving all the new logic (the following block) into a new method and turning the single-use helper methods (like `IsStaticEqualsMethod`) into local functions.